### PR TITLE
[mono] Use "dotnet publish" for Android sample with ILLink

### PR DESCRIPTION
**Pull request by EgorBo**
1) Use `-Os` opt level for mobile targets (it's size-friendly -O2 basically)
2) Use `-Wl,-s` for Android (uses strip util from NDK)
3) Hook into `dotnet publish` for Android sample, thus:
```
dotnet publish -c Release -r Android-arm64 /p:PublishTrimmed=true
```
Generates a minimal hello world APK (1.8 Mb) using in-tree runtime pack.

Here is extracted content of that APK:
image

Also, I think it makes sense to revise different unrolled algorithms like SpanHelpers, etc.

### CODE REVIEW

File: tools-local/tasks/mobile.tasks/AppleAppBuilder/Templates/CMakeLists.txt.template [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.14.5)
+cmake_minimum_required(VERSION 3.16)
```
Comment (EgorBo) [inline tools-local/tasks/mobile.tasks/AppleAppBuilder/Templates/CMakeLists.txt.template:1]: small unrelated change - we need 3.16 at least for proper iOS support.

**Review by steveisok — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]
